osutil/disks/disks_linux.go

```diff
@@ -507,46 +540,64 @@ func (d *disk) populatePartitions() error {
 	return nil
 }
 
-func (d *disk) FindMatchingPartitionUUIDWithPartLabel(label string) (string, error) {
+func (d *disk) FindMatchingPartitionWithPartLabel(label string) (Partition, error) {
```

Comment:
There's always this awkward question whether if something is not found the function should return an empty struct or a nil pointer. Since we switched to a struct here, perhaps *Partition is a better fit here and in the other Find function?

Also the disk lookup things would return a Disk interface which can be nil.

Reply:
sure returning a pointer seems reasonable

Reply:
ah this actually becomes somewhat awkward both in tests and in practice since Partitions() still returns a list of bare structs, so I think if we wanted to change to pointers here, we should change to pointers everywhere. I'm happy to do so, but it feels a bit like busy work for not much gain considering that these methods return concrete inspectable error types for the not found error so it's easy for callers to be sure that the partition was not found. So unless someone else feels the same way I'd rather leave it as-is for now

Reply:
That's fair, we'll get a plenty of opportunities to refactor this if needed.